ios/Trifle/Sources/Trifle.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  Trifle.swift
+//  Trifle
+//
+
+import Foundation
+
+public class Trifle {
+    
+    public static let version = "0.1.3"
+    
+    private let contentSigner: ContentSigner
+    
+    public init(
+        tag: String,
+        with level: Level = .secureEnclave
+    ) {
+        switch level {
+        case .secureEnclave:
+            self.contentSigner =
+            SecureEnclaveDigitalSignatureKeyManager(tag: tag)
+        }
+    }
+    
+    /**
+     Create a new mobile Trifle keypair for which can be used to create a
+     certificate request and to sign messages. The library (Trifle) will
+     automatically try to choose the best algorithm and key type available on
+     this device.
+
+     - returns: An opaque Trifle representation, `KeyHandle`,
+        of the key-pair, which the client will need to store.
+     */
+    public func generateSigningKeyHandle() {
+        // TODO: IMPLEMENT
+    }
+    
+    /**
+     Generate a Trifle MobileCertificateRequest, signed by the provided
+     keyHandle, that can be presented to the Certificate Authority (CA) for
+     verification.
+     
+     - returns: An opaque Trifle representation
+        `MobileCertificateRequest` of the certificate request.
+     */
+    public func generateMobileCertificateRequest(
+        format: CSRFormat,
+        version: UInt32 = 0
```

Comment:
is there a reason these are exposed as arguments?  I would expect clients to not need to deal with this

Reply:
Depends. Since we're still in the mode of we can represent the CSR as anything, I figured parameterizing by type and have a control flow makes sense. Besides,the developer still have to specify something about the cert request (x500 names, extensions, etc)

Reply:
we can represent it as anything, but if we expose this to clients, then we no longer have the flexibility to change it.  If this is part of an internal function that we do not expect Trifle users to call, then this is fine, but my impression was that this is the part that is being exposed.  As for other areas: x500 names, etc., I think those can be auto-populated by us for now, e.g. name could be "Trifle+device-identifier"

Reply:
if we dont parameterize this, how are developers outside of cash be using this. for example `common name`. they all cant be pointing to `cash.app`